File: Tareas/tarea12.rb
puts "Azul Carolina Martínez Cabañas"
puts "1868435					 ITS"
puts "Periodo Enero-Junio 2020"
puts "Ing. Ismael Gutiérrez Pimentel"
puts "Serie C"
puts "Ingrese el número de la serie al que se quiere llegar: "
n = gets().to_i

contador = 0
positivos = 2
negativos = -3
actual = 1
varSwitcheable = true
for i in (1..n)
  if(varSwitcheable)then
    for y in (1..actual)
      if(contador>=n)then
      abort()
      else
      print("#{positivos} ")
      positivos += 2
      contador += 1
      end
    end
    varSwitcheable=false
    actual += 1
  else
    for y in (1..3)
      if(contador>=n)then
        abort()
      else
      print("#{negativos} ")
      negativos -= 2
      contador += 1
      end
    end
    varSwitcheable=true
  end
end

File: Tareas/tarea12.java
import javax.swing.*;
import java.util.ArrayList;

public class Tarea12 {
    public static void main(String[] args) {
        JOptionPane.showMessageDialog(null, "Azul Carolina Martínez Cabañas\n1868435 ITS\nPeriodo Enero-Junio 2020\nIng. Ismael Gutiérrez Pimentel");
        int n = Integer.parseInt(JOptionPane.showInputDialog("Serie C\nIngrese el número de la serie al que se quiere llegar: "));

        ArrayList<Integer> impresion = new ArrayList<>();
        int contador = 0, positivos = 2, negativos = -3, actual = 1;
        boolean varSwitcheable = true;
        for (int i = 1; i <=n ; i++) {
            if(varSwitcheable){
                for (int j = 1; j <= actual ; j++) {
                    if(contador>=n){
                        continue;
                    }else{
                        impresion.add(positivos);
                        positivos +=2;
                        contador +=1;
                    }
                }
                varSwitcheable = false;
                actual +=1;
            }else{
                for(int j = 1; j<=3; j++){
                    if(contador>=n){
                        continue;
                    }else{
                        impresion.add(negativos);
                        negativos -=2;
                        contador +=1;
                    }
                }
                varSwitcheable = true;
            }
        }
        String mensaje ="";
        for(Integer numero: impresion){
            mensaje += numero + " ";
        }
        JOptionPane.showMessageDialog(null, mensaje);
    }
}